Source/cmNinjaTargetGenerator.cxx

```diff
@@ -333,6 +333,49 @@ std::string cmNinjaTargetGenerator::ComputeIncludes(
   return includesString;
 }
 
+//result - vector of all libraries
+std::vector<std::string> cmNinjaTargetGenerator::ComputeLinkDepsForJSON(
+  const std::string& linkLanguage, const std::string& config,
+  bool ignoreType) const
+{
+  std::vector<std::string> result{};
+  // Static libraries never depend on other targets for linking.
```

Comment:
What do you mean? I think static libraries depend on both libraries and object files